dist/src/main/java/io/camunda/application/ProfilesStreamlining.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application;
+
+import com.google.common.collect.Sets;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.env.EnvironmentPostProcessor;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.env.MapPropertySource;
+
+public class ProfilesStreamlining implements EnvironmentPostProcessor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ProfilesStreamlining.class);
+
+  private static final Set<String> MERGING_PROFILES = Set.of(
+      Profile.OPERATE.getId(),
+      Profile.TASKLIST.getId(),
+      Profile.IDENTITY.getId(),
+      Profile.GATEWAY.getId()
+  );
+
+  public boolean isSkipProcessor() {
+    return Boolean.parseBoolean(System.getenv("SKIP_PROCESSOR"));
+  }
+
+  @Override
+  public void postProcessEnvironment(
+      final ConfigurableEnvironment environment,
+      final SpringApplication application) {
+    if (isSkipProcessor()) {
+      LOGGER.info("\uD83D\uDFE7 Skipping profiles streamlining processor");
+      return;
+    }
```

Comment:
This is just here for debugging purposes: it's not meant to be merged with the final project.

Same thing for the emojis I am using in the log lines: I am using them to quickly find the entries in the logs when I launch the application. Everything will be removed after the POC